pypeit/core/datacube.py

```diff
@@ -126,8 +257,63 @@ def dar_correction(wave_arr, coord, obstime, location, pressure, temperature, re
     return ra_diff, dec_diff
 
 
+def make_whitelight_fromref(all_ra, all_dec, all_wave, all_sci, all_wghts, all_idx, dspat,
+                            ref_filename, numfiles=None):
+    """ Generate a whitelight image of every input frame,
+    based on a reference image. Note the, the reference
+    image must have a well-defined WCS.
+
+    Args:
+        all_ra (`numpy.ndarray`_):
```

Comment:
Indicate shapes of all inputs as is our doc convention.

Reply:
Done